core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -460,7 +460,9 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
         |                              on one of the worker machines inside the cluster ("cluster")
         |                              (Default: client).
         |  --class CLASS_NAME          Your application's main class (for Java / Scala apps).
-        |  --name NAME                 A name of your application.
+        |  --name NAME                 A name of your application. In yarn-cluster mode the name
```

Comment:
There is currently no way to bring yarn-cluster and yarn-client modes under the same hood due to the reasons I've mentioned in the description of the pull-request. Do you think, that `--name` should override the name set by `SparkContext.setAppName`?

We can not be consistent without the removal of `SparkContext.setAppName` or without overriding it with `--name` if set.